src/Arcus.Security.Secrets.AzureKeyVault/KeyVaultSecretProvider.cs

```diff
@@ -35,19 +35,26 @@ public class KeyVaultSecretProvider : ISecretProvider
         /// </summary>
         /// <param name="authenticator">The requested authentication type for connecting to the Azure Key Vault instance</param>
         /// <param name="vaultConfiguration">Configuration related to the Azure Key Vault instance to use</param>
+        [Obsolete("Use other constructor with " + nameof(IKeyVaultAuthentication) + " instead")]
+#pragma warning disable 618
         public KeyVaultSecretProvider(IKeyVaultAuthenticator authenticator, IKeyVaultConfiguration vaultConfiguration)
+            : this(new CompatibleKeyVaultAuthentication(authenticator), vaultConfiguration) { }
```

Comment:
We should be consistent in our formatting
```suggestion
            : this(new CompatibleKeyVaultAuthentication(authenticator), vaultConfiguration)
            {
            }
```